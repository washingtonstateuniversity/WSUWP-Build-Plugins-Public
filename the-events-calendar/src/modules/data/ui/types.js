/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_DASHBOARD_DATE_TIME = `${ PREFIX_EVENTS_STORE }/SET_DASHBOARD_DATE_TIME`;
export const TOGGLE_DASHBOARD_DATE_TIME = `${ PREFIX_EVENTS_STORE }/TOGGLE_DASHBOARD_DATE_TIME`;

export const SET_DASHBOARD_PRICE = `${ PREFIX_EVENTS_STORE }/SET_DASHBOARD_PRICE`;

export const SET_VISIBLE_MONTH = `${ PREFIX_EVENTS_STORE }/SET_VISIBLE_MONTH`;
