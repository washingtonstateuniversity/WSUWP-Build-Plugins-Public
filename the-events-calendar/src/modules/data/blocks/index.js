/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
