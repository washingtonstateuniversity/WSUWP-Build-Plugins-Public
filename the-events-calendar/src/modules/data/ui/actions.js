/**
 * Internal dependencies
 */
import * as types from './types';
import { moment as momentUtil } from '@moderntribe/common/utils';

export const toggleDashboardDateTime = () => ( {
	type: types.TOGGLE_DASHBOARD_DATE_TIME,
} );

export const openDashboardDateTime = () => ( {
	type: types.SET_DASHBOARD_DATE_TIME,
	payload: {
		open: true,
	},
} );

export const closeDashboardDateTime = () => ( {
	type: types.SET_DASHBOARD_DATE_TIME,
	payload: {
		open: false,
	},
} );

export const openDashboardPrice = () => ( {
	type: types.SET_DASHBOARD_PRICE,
	payload: {
		open: true,
	},
} );

export const closeDashboardPrice = () => ( {
	type: types.SET_DASHBOARD_PRICE,
	payload: {
		open: false,
	},
} );

export const setVisibleMonth = ( visibleMonth ) => ( {
	type: types.SET_VISIBLE_MONTH,
	payload: {
		visibleMonth,
	},
} );

export const setInitialState = ( { get } ) => ( dispatch ) => {
	const start = get( 'start' );
	if ( ! start ) {
		return;
	}

	const month = momentUtil.toMoment( momentUtil.parseFormats( start ) ).startOf( 'month' ).toDate();
	dispatch( setVisibleMonth( month ) );
};
