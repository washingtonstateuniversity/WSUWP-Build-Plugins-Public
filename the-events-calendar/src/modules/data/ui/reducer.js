/**
 * External dependencies
 */
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	dashboardDateTimeOpen: false,
	dashboardPriceOpen: false,
	visibleMonth: moment().startOf( 'month' ).toDate(),
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DASHBOARD_DATE_TIME:
			return {
				...state,
				dashboardDateTimeOpen: action.payload.open,
			};
		case types.TOGGLE_DASHBOARD_DATE_TIME:
			return {
				...state,
				dashboardDateTimeOpen: ! state.dashboardDateTimeOpen,
			};
		case types.SET_DASHBOARD_PRICE:
			return {
				...state,
				dashboardPriceOpen: action.payload.open,
			};
		case types.SET_VISIBLE_MONTH:
			return {
				...state,
				visibleMonth: action.payload.visibleMonth,
			};
		default:
			return state;
	}
};
