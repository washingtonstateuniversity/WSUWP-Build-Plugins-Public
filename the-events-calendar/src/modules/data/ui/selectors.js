/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getUI = ( state ) => state.events.ui;

export const getDashboardDateTimeOpen = createSelector(
	[ getUI ],
	( ui ) => ui.dashboardDateTimeOpen
);

export const getDashboardPriceOpen = createSelector(
	[ getUI ],
	( ui ) => ui.dashboardPriceOpen
);

export const getVisibleMonth = createSelector(
	[ getUI ],
	( ui ) => ui.visibleMonth
);
