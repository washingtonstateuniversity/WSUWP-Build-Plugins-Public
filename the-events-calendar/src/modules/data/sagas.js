
/**
 * External dependencies
 */
import { store } from '@moderntribe/common/store';

/**
 * Internal dependencies
 */
import * as datetime from '@moderntribe/events/data/blocks/datetime';
import * as price from '@moderntribe/events/data/blocks/price';
import * as venue from '@moderntribe/events/data/blocks/venue';
import * as website from '@moderntribe/events/data/blocks/website';
import * as sharing from '@moderntribe/events/data/blocks/sharing';
import * as classic from '@moderntribe/events/data/blocks/classic';

export default () => [
	price.sagas,
	venue.sagas,
	website.sagas,
	sharing.sagas,
	classic.sagas,
	datetime.sagas,
].forEach( sagas => store.run( sagas ) );
