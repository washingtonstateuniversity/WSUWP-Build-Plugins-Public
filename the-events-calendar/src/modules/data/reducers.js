/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import blocks from './blocks';
import ui from './ui';
import search from './search';
import details from './details';

export default combineReducers( {
	blocks,
	ui,
	search,
	details,
} );

