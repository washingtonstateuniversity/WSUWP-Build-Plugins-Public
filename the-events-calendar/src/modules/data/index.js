/**
 * External dependencies
 */
import reducer from './reducers';

import { plugins } from '@moderntribe/common/data';
import { store } from '@moderntribe/common/store';
import * as blocks from './blocks';
import initSagas from './sagas';

export const initStore = () => {
	const { dispatch, injectReducers } = store;

	initSagas();
	dispatch( plugins.actions.addPlugin( 'events' ) );
	injectReducers( { events: reducer } );
};

export const getStore = () => store;

export { blocks };
