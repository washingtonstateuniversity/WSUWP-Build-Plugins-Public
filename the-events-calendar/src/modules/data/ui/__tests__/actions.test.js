/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import moment from 'moment';

/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/ui';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - UI actions', () => {
	it( 'Should toggle the dashboard', () => {
		expect( actions.toggleDashboardDateTime() ).toMatchSnapshot();
	} );

	it( 'Should open the dashboard', () => {
		expect( actions.openDashboardDateTime() ).toMatchSnapshot();
		expect( actions.openDashboardPrice() ).toMatchSnapshot();
	} );

	it( 'Should close the dashboard', () => {
		expect( actions.closeDashboardDateTime() ).toMatchSnapshot();
		expect( actions.closeDashboardPrice() ).toMatchSnapshot();
	} );

	it( 'Should set the visible month', () => {
		Date.now = jest.fn( () => '2018-07-01T05:00:00.000Z' );
		expect( actions.setVisibleMonth( Date.now() ) ).toMatchSnapshot();
	} );

	it( 'Should not set the initial state', () => {
		const store = mockStore( {} );
		const get = jest.fn();
		store.dispatch( actions.setInitialState( { get } ) );

		expect( get ).toHaveBeenCalledTimes( 1 );
		expect( store.getActions() ).toEqual( [] );
	} );
} );

describe( '[STORE] - UI thunk actions', () => {
	it( 'Should set the initial state', () => {
		const store = mockStore( {} );
		const get = jest.fn( () => moment( '2018-07-01T00:00:00.000Z' ) );

		store.dispatch( actions.setInitialState( { get } ) );

		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 1 );
		expect( store.getActions() ).toMatchSnapshot();
	} );
} );
