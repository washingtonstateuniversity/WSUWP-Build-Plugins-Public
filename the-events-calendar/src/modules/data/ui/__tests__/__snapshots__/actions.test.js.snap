// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - UI actions Should close the dashboard 1`] = `
Object {
  "payload": Object {
    "open": false,
  },
  "type": "@@MT/EVENTS/SET_DASHBOARD_DATE_TIME",
}
`;

exports[`[STORE] - UI actions Should close the dashboard 2`] = `
Object {
  "payload": Object {
    "open": false,
  },
  "type": "@@MT/EVENTS/SET_DASHBOARD_PRICE",
}
`;

exports[`[STORE] - UI actions Should open the dashboard 1`] = `
Object {
  "payload": Object {
    "open": true,
  },
  "type": "@@MT/EVENTS/SET_DASHBOARD_DATE_TIME",
}
`;

exports[`[STORE] - UI actions Should open the dashboard 2`] = `
Object {
  "payload": Object {
    "open": true,
  },
  "type": "@@MT/EVENTS/SET_DASHBOARD_PRICE",
}
`;

exports[`[STORE] - UI actions Should set the visible month 1`] = `
Object {
  "payload": Object {
    "visibleMonth": "2018-07-01T05:00:00.000Z",
  },
  "type": "@@MT/EVENTS/SET_VISIBLE_MONTH",
}
`;

exports[`[STORE] - UI actions Should toggle the dashboard 1`] = `
Object {
  "type": "@@MT/EVENTS/TOGGLE_DASHBOARD_DATE_TIME",
}
`;

exports[`[STORE] - UI thunk actions Should set the initial state 1`] = `
Array [
  Object {
    "payload": Object {
      "visibleMonth": 2018-07-01T00:00:00.000Z,
    },
    "type": "@@MT/EVENTS/SET_VISIBLE_MONTH",
  },
]
`;
